app/lib/feed_manager.rb

```diff
@@ -95,31 +85,47 @@ def redis
   end
 
   def filter_from_home?(status, receiver)
-    should_filter = false
+    should_filter = receiver.muting?(status.account_id)                       # Filter if I'm muting this person
```

Comment:
If this is true, the method can return instantly, no need to go through all the other checks